frontend/src/components/IsochronesMap.vue

```diff
@@ -0,0 +1,172 @@
+
+watch(
+  () => props.selectedTransportMode,
+  () => {
+    fetchIsochrone(isochroneMarker.getLngLat());
+  }
+);
+
+type SourceNewAPI = {
+  setData: (data: FeatureCollection) => void;
+};
+
+type ApiError = {
+  error_description: string;
+};
+
+function fetchIsochrone(location: LngLatLike) {
+  // Use getIsochrone to fetch isochrone at the given location, for [1, 5, 10, 15] times
+  //Then set the geojson data to the source "isochrone" using the setData method
+
+  getIsochrone(
+    LngLat.convert(location).toArray() as [number, number],
+    props.selectedTransportMode,
+    [5, 10, 15].map((minutes) => minutes * 60)
+  )
+    .then((data: Feature<Geometry, GeoJsonProperties>[]) => {
+      if (map === null) return;
+      const source = map?.getSource("isochrone") as Source & SourceNewAPI;
+      source.setData({ type: "FeatureCollection", features: data });
+    })
+    .catch((err: AxiosError & ApiError) => {
+      error.value = true;
+      errorMessage.value = err.message + " : " + err.error_description;
+    });
+}
+
+function onGeocodingSearchResult(e: { result: { center: LngLatLike } }) {
+  if (map === null) return;
+  isochroneMarker.setLngLat(e.result.center).addTo(map);
+  map.flyTo({
+    center: e.result.center,
+    zoom: 12,
+  });
+
+  fetchIsochrone(e.result.center);
+}
+
+onMounted(() => {
+  map = new Map({
+    container: container.value as HTMLDivElement,
+    // nginx will redirect to the correct maptiler url adding the API key
+    style: "/maptiler/maps/basic-v2-light/style.json",
+    zoom: 7,
+    center,
+  }) as InstanceType<typeof Map>;
+
+  map.on("load", function () {
+    loading.value = false;
+
+    if (map == null) return;
+
+    map.addSource("isochrone", {
+      type: "geojson",
+      data: { type: "Feature", geometry: { type: "Polygon", coordinates: [] } },
+    });
+    // Add a new layer to visualize the isochrone.
+    map.addLayer({
+      id: "isochrone-fill",
+      type: "fill",
+      source: "isochrone", // reference the data source
+      layout: {},
+      paint: {
+        "fill-color": "#0080ff", // blue color fill
+        "fill-opacity": 0.5,
+      },
+    });
+
+    //Add layer for isochrone feature
+    map.addLayer({
+      id: "isochrone",
+      type: "fill",
+      source: "isochrone",
+      layout: {},
+      paint: {
+        "fill-color": [
+          "interpolate",
+          ["linear"],
+          ["get", "value"],
+          ...stepsColors(300, 900, mapColors),
+        ],
+        "fill-opacity": 0.5,
+      },
+    });
+
+    // This control is used to search for a location
+    map.addControl(geocoder.on("result", onGeocodingSearchResult));
+
+    geocoder.query("EPFL Lausanne");
+
+    isochroneMarker.on("dragend", () => {
+      const { lng, lat } = isochroneMarker.getLngLat();
+      fetchIsochrone([lng, lat]);
+    });
+  });
+});
+</script>
+
+<style scoped>
+.map {
+  min-height: 1000px;
+  width: 100%;
+  position: relative;
+}
+.v-alert {
```

Comment:
Maybe choose a custom class name ?

Reply:
@pierreleripoll sorry, now I better understand why you need this. ^^
Did you check for an existing class ? For instance https://vuetifyjs.com/en/components/grids/#align

Reply:
@blueur last commit I changed to use vuetify's API only, but since you mention it I couldn't find a way to fill available height with my map with vuetify's API so I add to use a main.css file